### PR TITLE
Tribits, Amesos: Remove spaces around time.h, allow superlu_dist without parmetis

### LINKED ISSUE #9094: tribits and amesos
## Enhancement
@trilinos/\<tribits\>

Remove spaces around time.h in gettimeofday.c as disliked by llvm on Windows.

@trilinos/\<amesos\>

Allow configuring with superludist but without parmetis, as superludist can now be built without parmetis

A PR is in the works.

### PR DESCRIPTION
**Pull request by jrobcary**
In tribits/win_interface/include/gettimeofday.c:

Remove spaces around time.h in its include, which spaces clang some version did not
like.

In packages/amesos/CMakeLists.txt:
With TPL_ENABLE_SuperLUDist_Without_ParMETIS=true, one can now
configure with superlu_dist and without parmetis - for the case where
superlu_dist is built without parmetis as is now possible with
-Denable_parmetislib:BOOL='OFF' when configuring superlu_dist.

@trilinos/<Tribits>
@trilinos/<Amesos>

These changes allow compilation of gettimeofday.c on Windows with LLVM,
and they allow building of Trilinos with superlu_dist and without parmetis, as
superlu_dist can now be built without parmetis.

* Closes #9094

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
@trilinos/amesos

File: packages/amesos/CMakeLists.txt
Comment (bartlettroscoe) [inline packages/amesos/CMakeLists.txt:13]: Since this option is really specific to Amesos, this should be called something like `Amesos_Allow_SuperLUDist_Without_ParMETIS` and it should be defined as a documented advanced cache var like:

```
ADVANCED_SET(Amesos_Allow_SuperLUDist_Without_ParMETIS OFF
  CACHE BOOL "Allow Amesos to use the SuperLUDist TPL without the enable of the ParMETIS TPL")
```

and then replace `TPL_ENABLE_SuperLUDist_Without_ParMETIS` with `Amesos_Allow_SuperLUDist_Without_ParMETIS` in this if statement.  (In which case you should loose the above comment.)

The name  `TPL_ENABLE_SuperLUDist_Without_ParMETIS` is too close to `TPL_ENABLE_SuperLUDist` which makes it seem this this is a TriBITS-supported var which it is not.  It is specific to Amesos and therefore should be prefixed with `Amesos`.

File: cmake/tribits/win_interface/include/gettimeofday.c
Comment (bartlettroscoe) [inline cmake/tribits/win_interface/include/gettimeofday.c:1]: This is fine obviously.  This is mirrored in https://github.com/TriBITSPub/TriBITS/pull/371.  It should be fine to commit this to Trilinos 'develop'.  I will take care of any merge conflict with the next update of TriBITS to Trilinos 'develop'.

**Review by bartlettroscoe — CHANGES_REQUESTED** [review]
I would just request that this add:

```
ADVANCED_SET(Amesos_Allow_SuperLUDist_Without_ParMETIS OFF
  CACHE BOOL "Allow Amesos to use the SuperLUDist TPL without the enable of the ParMETIS TPL")
```

and then replace `TPL_ENABLE_SuperLUDist_Without_ParMETIS` with `Amesos_Allow_SuperLUDist_Without_ParMETIS` in this if statement.  (In which case you should loose the comment above the if statement.)

**Review by bartlettroscoe — APPROVED** [review]
Looks great to me.  This maintains backward compatibility and nothing changes unless the user explicitly sets `-D Amesos_Allow_SuperLUDist_Without_ParMETIS=ON`.

But we should have someone from @trilinos/amesos review and approve this as well before the merge.

**Comment by bartlettroscoe** [comment]
Given this is a minor change to amesos/CMakeLists.txt file, I will go ahead and allow to merge without approval from someone from @trilinos/amesos.  (But in general package maintainers should review and approve every change to their package.)

**Review by ndellingwood — APPROVED** [review]

**Pull request merged**
